### PR TITLE
feat(CI) adding container labels, unit testing, and linting

### PR DESCRIPTION
**Pull request by DannyMassa**
Adding test and lint checks to inside the Docker container for portability and to offer an easy way of running checks locally. Also adding opencontainer labels to the container.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    name: Build Docker Image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Get short SHA
+        id: sha
+        run: |
+          echo ::set-output name=sha7::$(git rev-parse --short=7 ${{ github.sha }})
+      - name: Get current time
```
Comment (intlabs) [inline .github/workflows/ci.yml:18]: im a bit uncomfortable with the import here, is there no other action, or even just a run step that we can use here?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Build Without Pushing
+        uses: docker/build-push-action@v2
+        with:
+          push: false
+          tags: |
+            quay.io/dannymassa/testrepo:${{ steps.sha.outputs.sha7 }}
```
Comment (intlabs) [inline .github/workflows/ci.yml]: needs to be updated to the correct repo

**Review by intlabs — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (DannyMassa) [reply .github/workflows/ci.yml:18]: I'll spend more time searching. If there isn't a built-in from GitHub for this, I'll update it to use something built into ubuntu.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (DannyMassa) [reply .github/workflows/ci.yml]: ack

File: .github/workflows/ci.yml (hunk 1 above)
Comment (DannyMassa) [reply .github/workflows/ci.yml:18]: no reason to bring in more platform specific steps. accomplished the same thing with ubuntu's date

**Review by intlabs — APPROVED** [review]

**Pull request merged**